### PR TITLE
Add check parsetree is not NULL at _SPI_execute_plan and _SPI_prepare_plan

### PR DESCRIPTION
**Pull request by red1452**
Add check parsetree is not NULL at _SPI_execute_plan and _SPI_prepare_plan

The parsetree pointer may be NULL. It uses to get information about the node
without checking on NULL. At the other places at the _SPI_execute_plan function
and the _SPI_prepare_plan function this pointer is checked before using. Needed
check was added.

### CODE REVIEW

File: src/backend/executor/spi.c [hunk 1]
```diff
@@ -1982,7 +1982,7 @@ _SPI_prepare_plan(const char *src, SPIPlanPtr plan)
 		CompleteCachedPlan(plansource,
 						   stmt_list,
 						   NULL,
-						   nodeTag(parsetree),
+						   parsetree ? nodeTag(parsetree) : T_Invalid,
```
Comment (RekGRpth) [inline src/backend/executor/spi.c]: Your solution does not solve the `CreateCommandTag(parsetree)` problem above, while [mine](https://github.com/arenadata/gpdb/pull/732/files#diff-eca689b02874d585dbe0547b5b5af6e723a88247c89b662fa0f601f3d363d4f5R1950) does!

**Comment by RekGRpth** [comment]
add title to PR description

File: src/backend/executor/spi.c [hunk 2]
```diff
@@ -1935,6 +1935,8 @@ _SPI_prepare_plan(const char *src, SPIPlanPtr plan)
 		List	   *stmt_list;
 		CachedPlanSource *plansource;
 
+		Assert(parsetree);
```
Comment (RekGRpth) [inline src/backend/executor/spi.c:1938]: This doesn't solve anything at all! I corrected my https://github.com/arenadata/gpdb/pull/732 and now it solves both of your problems.

**Comment by red1452** [comment]
> add title to PR description

done.

**Pull request closed without merging**